src/main/java/uk/gov/hmcts/reform/sscs/service/ServiceHearingsService.java

```diff
@@ -47,20 +51,54 @@ public ServiceHearingValues getServiceHearingValues(ServiceHearingRequest reques
         String originalCaseData = objectMapper.writeValueAsString(caseData);
 
         ServiceHearingValues model = ServiceHearingValuesMapping.mapServiceHearingValues(caseData, referenceDataServiceHolder);
+        Optional<UnavailabilityRange> unavailabilityRangeWithDateProblems = model.getParties().stream()
+            .map(PartyDetails::getUnavailabilityRanges)
+            .flatMap(o -> o.stream()
+                .filter(p -> p.getUnavailableFromDate() != null && p.getUnavailableToDate() != null && p.getUnavailableToDate().isBefore(p.getUnavailableFromDate()))
```

Comment:
I'd prefer o and p to be replaced with something more descriptive - hard to understand what they are when I don't know the object being streamed.

Also please put each && on a new line.